common/src/main/java/tachyon/worker/block/io/LocalFileBlockWriter.java

```diff
@@ -67,7 +56,12 @@ public WritableByteChannel getChannel() {
 
   @Override
   public long append(ByteBuffer inputBuf) throws IOException {
-    return write(mLocalFileChannel.size(), inputBuf);
+    return write(mLocalFileChannel.size(), inputBuf.duplicate());
```

Comment:
So the `inputBuf` doesn't get modified during `append. Otherwise as `append` succeeds, the `inputBuf` also modifies its position pointer. e.g., if you do 
```java
writer.append(buf);
writer.append(buf);
```
the second call will append nothing as buf's reaching its limit after first call. Do you think this is what we expected?